openedx/features/job_board/views.py

```diff
@@ -15,6 +17,57 @@ class JobListView(ListView):
     ordering = ['-created']
     template_engine = 'mako'
 
+    def get_queryset(self):
+        self.param_job_query = self.request.GET.get('query')
+        self.param_job_type = self.request.GET.getlist('type')
+        self.param_job_city = self.request.GET.get('city')
+        self.param_job_country = self.request.GET.get('country')
+        self.param_job_hours = self.request.GET.getlist('hours')
+        self.param_job_compensation = self.request.GET.getlist('compensation')
+
+        country_codes = []
+        if self.param_job_country:
+            country_codes = [country[0] for country in countries if
+                             self. param_job_country.lower() in country[1].lower()]
+
+        queryset = super(JobListView, self).get_queryset()
+
+        if self.param_job_type:
+            queryset = queryset.filter(type__in=self.param_job_type)
+        if self.param_job_hours:
+            queryset = queryset.filter(hours__in=self.param_job_hours)
+        if self.param_job_compensation:
+            queryset = queryset.filter(compensation__in=self.param_job_compensation)
+        if country_codes:
+            queryset = queryset.filter(country__in=country_codes).filter(country__in=country_codes)
+        if self.param_job_city:
+            queryset = queryset.filter(city__icontains=self.param_job_city)
+        if self.param_job_query:
+            queryset = queryset.filter(title__icontains=self.param_job_query)
```

Comment:
@qasimgulzar please note that we have to compare some fields with single string values such as country and city, while other fields with a list of values using `{field}__in=list` like job type. We are also using case insensitive search in case of `city` and `query`

Reply:
But aren't django queries lazy and we'll still be querying the database just once in both the cases? Just trying to understand why this approach might be more optimised?

Reply:
@abdullah998 it is some of the code styles. it is just nice to have if you don't want to incorporate its fine.